### ISSUE TITLE
Reference counts and errors not working

### ISSUE THREAD

**Issue by Beider** [dotnet/vscode-csharp#6031]
Type: <b>Bug</b>

In my code base after the update a lot of reference counts are not shown anymore. In addition sometimes when I remove constants or variables it will not even show errors for the places where the constant/varible is still in use.

In addition there is an error reported "Request codeAction/resolve failed."

```
[Error - 10:26:05 AM] [LanguageServerHost]System.InvalidCastException: Unable to cast object of type 'Microsoft.CodeAnalysis.CSharp.Syntax.MemberAccessExpressionSyntax' to type 'Microsoft.CodeAnalysis.CSharp.Syntax.LocalDeclarationStatementSyntax'.
   at Microsoft.CodeAnalysis.CSharp.InlineDeclaration.CSharpInlineDeclarationCodeFixProvider.FixAllAsync(Document document, ImmutableArray`1 diagnostics, SyntaxEditor editor, CodeActionOptionsProvider fallbackOptions, CancellationToken cancellationToken) in /_/src/Analyzers/CSharp/CodeFixes/InlineDeclaration/CSharpInlineDeclarationCodeFixProvider.cs:line 61
   at Microsoft.CodeAnalysis.CodeFixes.SyntaxEditorBasedCodeFixProvider.FixAllWithEditorAsync(Document document, Func`2 editAsync, CancellationToken cancellationToken) in /_/src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeFixes/SyntaxEditorBasedCodeFixProvider.cs:line 84
   at Microsoft.CodeAnalysis.CodeActions.CodeAction.GetChangedSolutionAsync(CancellationToken cancellationToken) in /_/src/Workspaces/Core/Portable/CodeActions/CodeAction.cs:line 259
   at Microsoft.CodeAnalysis.CodeActions.CodeAction.ComputeOperationsAsync(CancellationToken cancellationToken) in /_/src/Workspaces/Core/Portable/CodeActions/CodeAction.cs:line 230
   at Microsoft.CodeAnalysis.CodeActions.CodeAction.ComputeOperationsAsync(IProgressTracker progressTracker, CancellationToken cancellationToken) in /_/src/Workspaces/Core/Portable/CodeActions/CodeAction.cs:line 242
   at Microsoft.CodeAnalysis.CodeActions.CodeAction.GetOperationsCoreAsync(Solution originalSolution, IProgressTracker progressTracker, CancellationToken cancellationToken) in /_/src/Workspaces/Core/Portable/CodeActions/CodeAction.cs:line 194
   at Microsoft.CodeAnalysis.LanguageServer.Handler.CodeActionResolveHandler.HandleRequestAsync(CodeAction codeAction, RequestContext context, CancellationToken cancellationToken) in /_/src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionResolveHandler.cs:line 84
   at Microsoft.CommonLanguageServerProtocol.Framework.QueueItem`3.StartRequestAsync(TRequestContext context, CancellationToken cancellationToken) in /_/src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/QueueItem.cs:line 176
[Error - 10:26:05 AM] Request codeAction/resolve failed.
  Message: Unable to cast object of type 'Microsoft.CodeAnalysis.CSharp.Syntax.MemberAccessExpressionSyntax' to type 'Microsoft.CodeAnalysis.CSharp.Syntax.LocalDeclarationStatementSyntax'.
  Code: -32000 
```

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i9-9900 CPU @ 3.10GHz (16 x 3096)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|15.88GB (6.01GB free)|
|Process Argv|--crash-reporter-id 86498245-2a6b-482f-ac2b-58471b684130|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
vserr242cf:30382550
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263cf:30335440
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
pythonvs932:30410667
py29gd2263cf:30792227
vsclangdc:30486549
c4g48928:30535728
dsvsc012cf:30540253
pynewext54:30695312
azure-dev_surveyone:30548225
vscccc:30803845
282f8724:30602487
89544117:30613380
vscrp:30673768
showlangstatbar:30737416
vsctsb:30748421
03d35959:30757346
pythonfmttext:30731395
pythoncmvfstrcf:30756944
fixshowwlkth:30771522
showindicator:30805244
pythongtdpath:30769146
i26e3531:30792625
gsofa:30804715
pythonnosmt12:30797651
pythonidxptcf:30805731
pythonnoceb:30805159
e537b577:30795824
dsvsc013:30795093
dsvsc014:30804076

```

</details>

**Comment by arkalyanms** [comment]
@Beider the extension version captured in the issue template seems to be on 1.26.0. Is that accurate?

**Comment by Beider** [comment]
Sorry my mistake, the second error I posted was accidentally taken from when I reverted after the new version had issues. I removed this now, the error in the post is from the last version that came out 1 day ago.

**Comment by dibarbet** [comment]
@Beider could you set the `dotnet.server.trace` setting to `Trace` and attach the C# output window logs?  That might help get an idea of where the code lens / diagnostics are going missing.

Also is there a particular file that triggers the code action resolve error?  If would you be able to share it?   Looks like there might be some specific syntax that the `CSharpInlineDeclarationCodeFixProvider` is not expecting.

assigning to @Cosifne to take a look at the code action resolve error.

**Comment by Cosifne** [comment]
Looked at the code and we have the type check in the analyzer https://github.com/dotnet/roslyn/blob/3e516fcb5478235ed85fec40803fa3676fa8ff4d/src/Analyzers/CSharp/Analyzers/InlineDeclaration/CSharpInlineDeclarationDiagnosticAnalyzer.cs#L149

My guess: If there are two requests in the request queue. If the request queue is processing the first time-consuming request, the document can get changed during the execution. Then the next request in the queue would start using the new document (I think we just rely on the URL, there are no version stamps), then the diagnostics data would be out-off date since the document has changed.

@Beider It would be great if you could share the log with us so that we can identify the problem.

**Comment by sharwell** [comment]
> In my code base after the update a lot of reference counts are not shown anymore. 

This is tracked as dotnet/roslyn#69583

**Comment by Beider** [comment]
Sorry I forgot about this,

I retried with the latest version again now. I also made sure to update everything else (dotnet, etc..) and now it seems to work fine. So I am not certain if the problem was just in version 2.0.320 or perhaps something else I was using was of an older version.

Eitherway, everything is working fine for me now so this can probably be closed if no one else has the same problems.
